### PR TITLE
docs: update docs that cite compose files

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

1. removed version marker (obsolete) and not used in our default Compose file
2. updated `domainname: mail` => `hostname: mail.*`

This PR may be merged at any time.

## Type of change

- [x] This change is a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: docs/content/config/security/ssl.md [hunk 1]
```diff
@@ -27,43 +27,29 @@ After installation, you can test your setup with:
 
 An [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) (_Fully Qualified Domain Name_) such as `mail.example.com` is required for `docker-mailserver` to function correctly, especially for looking up the correct SSL certificate to use.
 
-Internally, `hostname -f` will be used to retrieve the FQDN as configured in the below examples.
+- `mail.example.com` will still uses `user@example.com` as the mail address.
```
Comment (polarathene) [inline docs/content/config/security/ssl.md:30]:
```suggestion
- `mail.example.com` will still use `user@example.com` as the mail address. You do not need a bare domain for that.
```

File: docs/content/config/security/ssl.md [hunk 2]
```diff
@@ -27,43 +27,29 @@ After installation, you can test your setup with:
...
+- We usually advise not to use a bare domain (_DNS MX record does not point to a subdomain_) for your `docker-mailserver`, but [just `example.com` is supported][docs-faq-baredomain].
```
Comment (polarathene) [inline docs/content/config/security/ssl.md:31]:
```suggestion
- We usually discourage assigning a bare domain (_When your DNS MX record does not point to a subdomain_) to represent `docker-mailserver`. However, an FQDN of [just `example.com` is also supported][docs-faq-baredomain].
```

**Review by polarathene — CHANGES_REQUESTED** [review]

**Review by polarathene — APPROVED** [review]
LGTM :+1: 

I disabled auto-merge due to additional changes I made, you may want to briefly review them. Commits are either descriptive by title or provide a little bit more context in commit messages.

**Comment by georglauterbach** [comment]
I like your suggestions 👍🏼

**Pull request merged**